components/core/src/util/text_render.rs

```diff
@@ -0,0 +1,37 @@
+use crate::error::{Error,
+                   Result};
+use std::io::Write;
+
+use serde::Serialize;
+use serde_json::Value as Json;
+use tabwriter::TabWriter;
+
+// Returns a library object that implements elastic tabstops
+pub fn tabw() -> TabWriter<Vec<u8>> { TabWriter::new(Vec::new()) }
+
+// Format strings with elastic tab stops
+pub fn tabify(mut tw: TabWriter<Vec<u8>>, s: &str) -> Result<String> {
+    write!(&mut tw, "{}", s)?;
+    tw.flush()?;
+    let res = tw.into_inner();
+    if res.is_err() {
+        return Err(Error::TabWriterIntoInnerFailed("Unable to flush \
+                                                    tabwriter buffer to \
+                                                    inner."
+                                                           .to_string()));
+    }
+    let inner = res.unwrap();
```

Comment:
I dont think we should have a custom `TabWriterIntoInnerFailed` type. Instead we could use the `io::Error` that is part of [`tabwriter::IntoInnerError`](https://docs.rs/tabwriter/1.2.1/tabwriter/struct.IntoInnerError.html#method.error) 

We could use:

```rust
    let inner = tw.into_inner().map_err(|e| {
                                    IoError::new(e.error().kind(),
                                                 "Unable to flush tabwriter buffer to inner.")
                                })?;
```

This preserves the `io::Error` kind, prevents us from having to create a new type, and avoids an unwrap.

Reply:
Modified as such in https://github.com/habitat-sh/habitat/pull/7841/commits/78f9d5e6a2b76e641bb6f2a2971c62744b5e762b